### DISCUSSION TITLE
Supply & Demand

### DISCUSSION THREAD

**Discussion by jimtollan** [DaveSkender/Stock.Indicators#1037] [Ideas]
Just a quick question (and possibly a suggestion ?!).

I've been looking at Supply & Demand (smart money concepts in Forex) recently and was wondering how to pull an indicator together that would draw these zones on a chart (and be useable for order placement also in code). This is basically  large range of support/resistance made visible and governed by a few rules:

1. We start by identifying the current price then we look to the left until we find a big strong move up or down
2. Then, we find any Extended Range Candle(s), (i.e. a candle that has a large body and small or no wicks)
3. Once we identify an ERC type of candles and we have either a drop or a rally in price, now we have to find the origin of that move (aka the base)
4. Once we identify the base, we can begin to draw our zone

The image below shows this far better than my words:

![image](https://user-images.githubusercontent.com/87867061/222001984-efbfe5aa-bf15-446f-9057-79ef22a82d0b.png)

We then crystalise this identification into a set of ranges on the chart that looks like the below:

![image](https://user-images.githubusercontent.com/87867061/222002825-b0618cfa-c9d1-4029-b8fd-fc389e70879b.png)

Is there anything already present that does this or is there a set of indicator combos that anyone could recommend to create such an indicator??

For those not familiar with smc trading, this gives a pretty decent rundown on the approaches: 

- https://tradingfxhub.com/blog/the-ultimate-guide-to-master-supply-and-demand-in-forex/ 
- https://www.flowbank.com/en/research/price-action-trading-strategy-supply-demand-zones
- https://smt-fx.com/smc-guide/

Hope this sparks a few thoughts ;)

jim

**Comment by DaveSkender** [comment]
I can’t recommend and overall approach here, but a candlestick pattern we’ve not yet implemented might be helpful: [long body](https://school.stockcharts.com/doku.php?id=chart_analysis:candlestick_pattern_dictionary#long_body_long_day) pattern.  The other aspect is to identify one of the several types of base patterns, one might be [pivots](https://dotnet.stockindicators.dev/indicators/Pivots/#content) with a wide span.

**Reply by jimtollan** [reply]
Thanks Dave - I'll review those options and update the thread, should they work out (or even if they don't!! :))

cheers again
jim
